File: dsa-cp/selection_sort.js
/*
selection sort algorithm with javascript
*/
function selectionSort(array){

	for(var i=0;i<array.length;i++){
		let min=i;
		for(var j=i+1;j<array.length;j++){
			if(array[j]<array[min]){
				min=j;
			}
		}
		if(i!==min){
			temp=array[i]
			array[i]=array[min]
			array[min]=temp
		}
	}
	return array;
}

var arr = [0,4,2,8,5,9,1,3,6,7];
const result = selectionSort(arr)
console.log(result)

File: java/dsa/selection-sort.java
// selection sort in java
class SelectionSort{
    public static void approce1(int arr[]){
        for(int i=0;i<arr.length-1;i++){
            int smallest = i;
            for(int j=i+1;j<arr.length;j++){
                if(arr[smallest]>arr[j]){
                    smallest=j;
                }
            }
            int temp = arr[i];
            arr[i]=arr[smallest];
            arr[smallest]=temp;
        }
        // print array
        for(int i=0;i<arr.length;i++){
            System.out.println(arr[i]);
        }
    }
    public static void main(String arg[]){
        int[] array ={2,3,1,6,4,5};
        approce1(array);
    }
}